### PR TITLE
Allow black's Github action params overriding.

### PR DESCRIPTION
**Pull request by hadialqattan**
Fix: #1751

### CODE REVIEW

**Review by cooperlees — APPROVED** [review]
LGTM! Thanks - much more flexible.

**Comment by hadialqattan** [comment]
> LGTM! Thanks - much more flexible.

That's my pleasure 🌹

File: docs/github_actions.md [hunk 1]
```diff
@@ -15,5 +15,7 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
-      - uses: psf/black@stable
+      - uses: psf/black@stable # the default is equivalent to `black . --diff --check`.
+        with: # (optional - override the default params).
+          args: [".", "--diff", "--check"]
```
Comment (ichard26) [inline docs/github_actions.md:20]: I just copied and pasted this example config and it has a syntactical error: https://github.com/ichard26/testblackaction/actions/runs/312826169

Appears that `with.args` doesn't support arrays containing strings :/ 
> A string that defines the inputs for a Docker container. GitHub passes the args to the container's ENTRYPOINT when the container starts up. An array of strings is not supported by this parameter.

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepswithargs

**Review by ichard26 — CHANGES_REQUESTED** [review]
Hey thanks for the PR! Please fix the example though.

File: docs/github_actions.md (hunk 1 above)
Comment (hadialqattan) [reply docs/github_actions.md]: Sorry about that, updated.

**Comment by hadialqattan** [comment]
> Hey thanks for the PR! Please fix the example though.

That's my pleasure 🌹

Thanks a lot for the review!

**Review by ichard26 — CHANGES_REQUESTED** [review]
While the example configuration is now valid, the action is still not configurable. [Look at this run for example](https://github.com/ichard26/testblackaction/runs/1269727798?check_suite_focus=true), the workflow file contains this:

```yaml
     - uses: psf/black@stable # the default is equivalent to `black . --diff --check`.
       with: # (optional - override the default params).
         args: "--version"
```

Black should just print its version and then quit, but the logs shows otherwise. I have no idea why this is happening as when I copy the entrypoint script locally it works perfectly. Maybe Docker/GHA related?

**Comment by hadialqattan** [comment]
> While the example configuration is now valid, the action is still not configurable. [Look at this run for example](https://github.com/ichard26/testblackaction/runs/1269727798?check_suite_focus=true), the workflow file contains this:
> 
> ```yaml
>      - uses: psf/black@stable # the default is equivalent to `black . --diff --check`.
>        with: # (optional - override the default params).
>          args: "--version"
> ```
> 
> Black should just print its version and then quit, but the logs shows otherwise. I have no idea why this is happening as when I copy the entrypoint script locally it works perfectly. Maybe Docker/GHA related?

I don't think so, it's because we're using the stable version which has no customization option!

**Comment by ichard26** [comment]
Okay, good catch, but why does the action still fails to run after pointing your fork's action branch? https://github.com/ichard26/testblackaction/runs/1271524579?check_suite_focus=true

**Comment by hadialqattan** [comment]
> Okay, good catch, but after pointing to your fork's action branch the action still fails to run ... https://github.com/ichard26/testblackaction/runs/1271524579?check_suite_focus=true

OMG, I don't know why I didn't make it executable!!!

**Sorry** for that, **UPDATED**.

**Please see: https://github.com/hadialqattan/testblackaction/runs/1272182754?check_suite_focus=true**

**Review by ichard26 — APPROVED** [review]
It finally works! :tada: 
https://github.com/ichard26/testblackaction/runs/1272184851

Don't worry about us having to go back and forth a few times, it wasn't a big deal. Just take it as a lesson to test out your change before submitting a PR if possible :)

Thanks for being cooperative with me and making this a smooth review experience.

**Comment by hadialqattan** [comment]
> It finally works! 🎉
> https://github.com/ichard26/testblackaction/runs/1272184851
> 
> Don't worry about us having to go back and forth a few times, it wasn't a big deal. Just take it as a lesson to test out your change before submitting a PR if possible :)
> 
> Thanks for being cooperative with me and making this a smooth review experience.

I appreciate the fast response!

Thank you for your patience, greetings.

**Pull request merged**